From 89a52f35ac455f4ea8395433bf99dafb2a03c27c Mon Sep 17 00:00:00 2001
From: siddhesh <siddhesh>
Date: Fri, 10 Aug 2012 18:55:16 +0000
Subject: [PATCH] 	* python/py-type.c (convert_field): Use gdb_py_long_from_longest
 	for TYPE_FIELD_BITPOS.
 	(typy_get_sizeof): Likewise for TYPE_LENGTH.

---
 gdb/ChangeLog        |    6 ++++++
 gdb/python/py-type.c |    4 ++--
 2 files changed, 8 insertions(+), 2 deletions(-)

Index: gdb-7.2/gdb/python/py-type.c
===================================================================
--- gdb-7.2.orig/gdb/python/py-type.c
+++ gdb-7.2/gdb/python/py-type.c
@@ -166,7 +166,7 @@ convert_field (struct type *type, int fi
 	}
       else
 	{
-	  arg = PyLong_FromLong (TYPE_FIELD_BITPOS (type, field));
+	  arg = gdb_py_long_from_longest (TYPE_FIELD_BITPOS (type, field));
 	  attrstring = "bitpos";
 	}
 
@@ -458,7 +458,7 @@ typy_get_sizeof (PyObject *self, void *c
     }
   /* Ignore exceptions.  */
 
-  return PyLong_FromLong (TYPE_LENGTH (type));
+  return gdb_py_long_from_longest (TYPE_LENGTH (type));
 }
 
 static struct type *
